test/sql/copy/partitioned/hive_file_filter.test

```diff
@@ -0,0 +1,84 @@
+# name: test/sql/copy/partitioned/hive_file_filter.test
+# description: use ComplexFilterPushdown() to prune partitions and find files
+# group: [partitioned]
+
+require parquet
+
+# enable_verification runs into optimization errors
+# statement ok
+# PRAGMA enable_verification
+
+# create a table
+statement ok
+CREATE TABLE t(year INTEGER, month INTEGER, day INTEGER, val INTEGER);
+
+foreach _year 1 2
+
+foreach _month 1 2 3 4 5 6 7 8 9 10 11 12
+
+foreach _day 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
+
+statement ok
+INSERT INTO t VALUES(${_year},${_month},${_day},${_year}*${_month}*${_day});
+
+endloop
+
+endloop
+
+endloop
+
+query I
+SELECT COUNT(*) FROM t;
+----
+720
+
+# create hive partition
+statement ok
+COPY t TO '__TEST_DIR__/filter_partition' (FORMAT PARQUET, PARTITION_BY(year,month,day));
+
+query I
+SELECT COUNT(*) FROM GLOB('__TEST_DIR__/filter_partition/**');
+----
+720
+
+# check amount of files (_year * 12 * 30)
+query I
+SELECT COUNT(*) FROM read_parquet('__TEST_DIR__/filter_partition/*/*/*/*.parquet', hive_partitioning=1, hive_filter=1);
+----
+720
+
+query I
+SELECT COUNT(*) FROM read_parquet('__TEST_DIR__/filter_partition/**', hive_filter=1);
+----
+720
+
+query I
+SELECT COUNT(*) FROM read_parquet('__TEST_DIR__/filter_partition/**', hive_filter=1) where year=1 and month>=1 and month<=12;
+----
+360
+
+query I
+SELECT COUNT(*) FROM read_parquet('__TEST_DIR__/filter_partition/**', hive_filter=1) where year=1 and month=10 and day=16;
+----
+1
+
+query I
+SELECT val FROM read_parquet('__TEST_DIR__/filter_partition/**', hive_filter=1) where year=1 and month=10 and day=16;
+----
+160
+
+# check pattern matching
+query I
+SELECT COUNT(*) FROM read_parquet('__TEST_DIR__/filter_partition/year=*/month=*/day=*/*.parquet', hive_filter=1);
+----
+720
+
+query I
+SELECT COUNT(*) FROM read_parquet('__TEST_DIR__/filter_partition/**/*.parquet', hive_filter=1);
+----
+720
+
+query I
+SELECT COUNT(*) FROM read_parquet('__TEST_DIR__/filter_partition/*/mon?h=1?/day=?/*.parquet', hive_filter=1);
```

Comment:
Could you add some tests that confirm things fail relatively nicely?
E.g. when we run into malformed tables. There are a few examples data sets in data/parquet-testing that are malformed in different ways. Note that as an experimental feature it is okay that the error message may not yet be perfect, but we should still aim to have it covered by a test case to make it explicit.